js/ascendex.js

```diff
@@ -467,55 +483,101 @@ module.exports = class ascendex extends Exchange {
                 'price': this.safeNumber (market, 'tickSize'),
             };
             const status = this.safeString (market, 'status');
             const active = (status === 'Normal');
-            const type = ('useLot' in market) ? 'spot' : 'future';
+            const type = ('useLot' in market) ? 'spot' : 'swap';
             const spot = (type === 'spot');
-            const future = (type === 'future');
+            const perpetual = (type === 'swap');
             const margin = this.safeValue (market, 'marginTradable', false);
+            let contract = false;
+            if (margin || perpetual === true) {
+                contract = true;
+            }
+            let linear = false;
+            if (perpetual) {
+                linear = true;
+            }
+            let minQty = this.safeNumber (market, 'minQty');
+            let maxQty = this.safeNumber (market, 'maxQty');
+            let minPrice = this.safeNumber (market, 'tickSize');
+            let maxPrice = undefined;
             let symbol = id;
-            if (!future) {
+            if (!perpetual) {
                 symbol = base + '/' + quote;
+            } else {
+                const lotSizeFilter = this.safeValue (market, 'lotSizeFilter');
+                minQty = this.safeNumber (lotSizeFilter, 'minQty');
+                maxQty = this.safeNumber (lotSizeFilter, 'maxQty');
+                const priceFilter = this.safeValue (market, 'priceFilter');
+                minPrice = this.safeNumber (priceFilter, 'minPrice');
+                maxPrice = this.safeNumber (priceFilter, 'maxPrice');
+                const str = this.safeString (market, 'underlying');
+                const substrings = str.split ('/');
+                baseId = substrings[0];
+                quoteId = substrings[1];
+                base = this.safeCurrencyCode (baseId);
+                quote = this.safeCurrencyCode (quoteId);
+                symbol = base + '/' + quote + ':' + settle;
             }
             const fee = this.safeNumber (market, 'commissionReserveRate');
             result.push ({
                 'id': id,
                 'symbol': symbol,
                 'base': base,
                 'quote': quote,
+                'settle': settle,
                 'baseId': baseId,
                 'quoteId': quoteId,
-                'info': market,
+                'settleId': settleId,
                 'type': type,
                 'spot': spot,
                 'margin': margin,
-                'future': future,
+                'swap': perpetual,
+                'future': false,
+                'option': false,
                 'active': active,
-                'precision': precision,
+                'derivative': contract,
+                'contract': contract,
+                'linear': linear,
+                'inverse': undefined,
                 'taker': fee,
                 'maker': fee,
+                'contractSize': undefined,
```

Comment:
this maybe should default to 1, I'm not sure, do you know what this property is about?

Reply:
I'm not sure what this value is for, is it supposed to be related to options?

Reply:
A contract has some value or size, that is often denominated in the settlementCurrency.

Reply:
Is this in the docs that I can review? Do you mean the current market value of the contract? Or for options the contract is usually for 100 shares of the underlying asset in traditional finance, I think cryptocurrency exchanges usually use a different number than 100 because everything is fractional so is that what is meant?

Reply:
Most often each exchange will have a contract specification:

- https://ascendex.com/en/futures/intro/btc-perp

That information is served from their APIs and we have to fill it into markets.